packages/stark-ui/src/modules/table/components/table.component.ts

```diff
@@ -344,10 +344,13 @@ export class StarkTableComponent extends AbstractStarkUiComponent implements OnI
 		if (changes["data"] && !changes["data"].isFirstChange()) {
 			this.resetFilterValueOnDataChange();
 
-			// When data changes, we should sort it again in any case because:
-			//   * orderProperties could contain some sort rules
-			//   * every column could have a sort rule defined
-			this.sortData();
+			if (this.orderProperties instanceof Array && this.orderProperties.length) {
+				// TODO: refactor -> sortData() internally calls initializeDataSource() and applyFilter()
+				this.sortData();
+			} else {
+				this.initializeDataSource();
+				this.applyFilter();
```

Comment:
But if we look in `sortData()` code, we can there the following calls: 
```typescript
this.initializeDataSource();
this.applyFilter();
```

Do we agree that we should rewrite the implementation of `sortData` because it is currently wrong ?

Reply:
Indeed, we should just move out those calls. In fact we have these FIXME there ;) 

```
// FIXME: move these two calls out of this method
this.initializeDataSource();

this.applyFilter();
```

Reply:
Yup' ! 
I just wanted to be sure about this 😊